compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp

```diff
@@ -599,37 +599,41 @@ std::shared_future<void> ConfigurationAdminImpl::NotifyConfigurationUpdated(
     }
 
     const auto managedServiceWrappers = managedServiceTracker.GetServices();
-    const auto it = std::find_if(
-      std::begin(managedServiceWrappers),
-      std::end(managedServiceWrappers),
-      [&pid](const auto& managedServiceWrapper) {
-        // The ServiceTracker will return a default constructed shared_ptr for each ManagedService
-        // that we aren't tracking. We must be careful not to dereference these!
-        return (managedServiceWrapper ? (pid == managedServiceWrapper->pid)
-                                      : false);
-      });
-    if (it != std::end(managedServiceWrappers)) {
+    auto it = managedServiceWrappers.begin();
+    while (
+      (it = std::find_if(
+         it,
+         std::end(managedServiceWrappers),
+         [&pid](const auto& managedServiceWrapper) {
+           // The ServiceTracker will return a default constructed shared_ptr for each ManagedService
+           // that we aren't tracking. We must be careful not to dereference these!
+           return (managedServiceWrapper ? (pid == managedServiceWrapper->pid)
+                                         : false);
+         })) != managedServiceWrappers.end()) {
       const auto& managedServiceWrapper = *it;
       notifyServiceUpdated(
-        pid, *(managedServiceWrapper->trackedService), properties, *logger);
+          pid, *(managedServiceWrapper->trackedService), properties, *logger);
+       it++;
     }
```

Comment:
is there a better way to do this that uses the C++ standard algorithms (`#include <algorithm>`)? This is iterating over all managed services and also running `std::find_if` for each iteration. Using `std::for_each` eliminates the manual iterator manipulation and reduces the time complexity,.

Reply:
Replaced code with a for_each loop. Please review again when you get a chance.